### PR TITLE
feat(Kafka Node): Add support for Snappy compression

### PR DESCRIPTION
**Pull request by merryech0**
This PR adds support of Snappy compression for Kafka node.

### CODE REVIEW

File: docker/images/n8n-custom/Dockerfile [hunk 1]
```diff
@@ -17,7 +17,7 @@ RUN pnpm build
 RUN rm -rf node_modules
 RUN jq 'del(.pnpm.patchedDependencies)' package.json > package.json.tmp; mv package.json.tmp package.json
 RUN node scripts/trim-fe-packageJson.js
-RUN NODE_ENV=production pnpm install --prod --no-optional
```
Comment (ekho) [inline docker/images/n8n-custom/Dockerfile:20]: `--no-optional` flag prevents installing Snappy binary dependencies

**Comment by brokenjacobs** [comment]
Any chance of adding support for zstd in here too?

**Comment by c-arstensen** [comment]
> Any chance of adding support for zstd in here too?

Second this

**Comment by brokenjacobs** [comment]
zstd / snappy / lz4 / gzip are required for standard interoperability with kafka. I understand not including support in a custom consumer you are building as part of an application if it's not used in your organization. But given that this project is intended for generic use with Kafka, it should include support the standard compression types used by kafka.

**Comment by mikebrandl** [comment]
We would really appreciate this functionality being included too!

**Pull request closed without merging**